### PR TITLE
Fix queue pop error and stat empty repository error

### PR DESCRIPTION
**Pull request by lunny**
This PR will fix two errors.

### CODE REVIEW

File: modules/queue/queue_redis.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func (fifo *RedisByteFIFO) PushFunc(data []byte, fn func() error) error {
 // Pop pops data from the start of the fifo
 func (fifo *RedisByteFIFO) Pop() ([]byte, error) {
 	data, err := fifo.client.LPop(fifo.queueName).Bytes()
-	if err != nil && err == redis.Nil {
+	if err == nil || err == redis.Nil {
```
Comment (zeripath) [inline modules/queue/queue_redis.go:124]: I guess this change is just for consistency?

File: modules/queue/unique_queue_redis.go [hunk 2]
```diff
@@ -102,11 +104,7 @@ func (fifo *RedisUniqueByteFIFO) PushFunc(data []byte, fn func() error) error {
 // Pop pops data from the start of the fifo
 func (fifo *RedisUniqueByteFIFO) Pop() ([]byte, error) {
 	data, err := fifo.client.LPop(fifo.queueName).Bytes()
-	if err != nil {
-		return data, err
-	}
-
-	if len(data) == 0 {
+	if err == nil || err == redis.Nil {
```
Comment (zeripath) [inline modules/queue/unique_queue_redis.go]: Oops. Not sure how that sneaked in.

**Review by zeripath — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
ping lgtm

File: modules/queue/unique_queue_redis.go (hunk 2 above)
Comment (lunny) [reply modules/queue/unique_queue_redis.go]: OK. This is still not right. I sent another commit.

**Pull request merged**